### DISCUSSION TITLE
Support for appending to an existing target object

### DISCUSSION THREAD

**Discussion by TallBaldGeek** [DedAnton/NextGenMapper#150] [Ideas]
I was testing out NextGenMapper to see if it could replace some reflection-based mapping that we're currently doing in our application. It looks like a promising fit, but we have many use cases where we need to map several source objects to a single target object. 

For example, if the target comprised of data from multiple sources as follows:
```csharp
record SourcePhone(string PhoneNumber, string PhoneType);
record SourceName(string FirstName, string LastName);
record CustomerDto(string PhoneNumber, string PhoneType, string FirstName, string LastName);
```

It would be helpful to be able to build a complete CustomerDto by combining data from several sources:
```csharp
var phone = new SourcePhone("5556667777", "Mobile");
var name = new SourceName("First", "Last");
var customer = phone.Map<CustomerDto>();
customer.MapAppend(name);

//or possibly in a fluent syntax?
var customer = phone.Map<CustomerDto>().Append(name);
```

This feature would be similar to Mapperly's support for using an [existing target object](https://mapperly.riok.app/docs/configuration/existing-target/).

**Comment by DedAnton** [comment]
Hello! Thank you very much for sharing your idea.

The proposal is very interesting and it seems to me that such functionality would be useful, but it will take a lot of time to implement it, and I only have enough time to fix bugs.

I will create a task with your proposal and put it in the backlog, but unfortunately I can’t give at least some time frame when I can do it
